gt/main.c

```diff
@@ -757,6 +842,127 @@ alloc_and_fill_notify_pkt(unsigned int socket, struct ggu_policy *policy,
 	return notify_pkt;
 }
 
+static void
+print_unsent_policy(struct ggu_policy *policy)
+{
+	int ret;
+	char err_msg[1024];
+
+	if (policy->state == GK_REQUEST) {
+		ret = snprintf(err_msg, sizeof(err_msg),
+			"gt: failed to send out the notification to Gatekeeper with policy decision [state: %hhu]",
+			policy->state);
+	} else if (policy->state == GK_GRANTED) {
+		ret = snprintf(err_msg, sizeof(err_msg),
+			"gt: failed to send out the notification to Gatekeeper with policy decision [state: %hhu, tx_rate_kb_sec: %u, cap_expire_sec: %u, next_renewal_ms: %u, renewal_step_ms: %u]",
+			policy->state, policy->params.u.granted.tx_rate_kb_sec,
+			policy->params.u.granted.cap_expire_sec,
+			policy->params.u.granted.next_renewal_ms,
+			policy->params.u.granted.renewal_step_ms);
+	} else if (policy->state == GK_DECLINED) {
+		ret = snprintf(err_msg, sizeof(err_msg),
+			"gt: failed to send out the notification to Gatekeeper with policy decision [state: %hhu, expire_sec: %u]",
+			policy->state,
+			policy->params.u.declined.expire_sec);
+	} else {
+		RTE_LOG(WARNING, GATEKEEPER,
+			"gt: unknown policy decision with state %hhu at %s, there is a bug in the Lua policy!\n",
+			policy->state, __func__);
+		return;
```

Comment:
Generate `err_msg` as in the other cases since the IP addresses added later will bring valuable information to the log message.